connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -108,6 +110,27 @@ public ClassLoader currentThreadLoader() {
         return Thread.currentThread().getContextClassLoader();
     }
 
+    /*
+     * Retrieves the connector type by connector name
+     *
+     * @return the ConnectorType for the given connector name.
+     * @throws ConnectException if the ConnectorType cannot be determined
+     */
+    public ConnectorType connectorType(String clsName) {
+        ClassLoader savedLoader = currentThreadLoader();
+        try {
+            savedLoader = compareAndSwapLoaders(clsName);
+            try {
+                return SinkConnector.class.isAssignableFrom(Class.forName(clsName)) ?
```

Comment:
As I mentioned before, I think we also need to handle classname aliasing. I would recommend factoring the logic to load the `Connector` class out of `newConnector`

Reply:
@hachikuji's comment is straight to the point. 

Even if we are not introducing a hard-to-detect bug here but bypassing the normal code path for loading a connector class (which involves calling `newConnector` here and as you can see it utilizes the delegating class loader), we are hurting the maintainability of the code by introducing a second way to load connector classes. 

Some evidences are: 

- This is the only place where `forName` is used in Connect code base. This divergence is not necessary here and will be hard to maintain. 
- Utilizing `delegatingLoader` for loading a connector class is not immediately obvious. Even if this ends up to be the case via the successive calls to `compareAndSwapLoaders` and `connectorLoader` the fact that one has to jump several hoops to confirm this, reduces maintainability as well.
- Overall, the way this method is implemented does not save anything by first calling `forName` and then the actual `newConnector`. To the contrary makes missing an alias riskier, especially in the event of future extensions in classloading isolation. 

Finally, `AbstractHerder` has already a `getConnector` method. Taking @hachikuji's suggestion a step further, I'd suggest something very simple as: 

```java
    protected ConnectorType connectorType(String connType) {
        return ConnectorType.from(getConnector(connType).getClass());
    }
```

to be added to `AbstractHerder` class instead of `Plugins`. The above is preferrable because it reuses the functionality already present for loading a connector class, uses the simple caching scheme present in `getConnector`, and reuses the `isAssignable` logic that is already present in `ConnectorType`. By the way, this last check is necessary because the code currently is not correct assuming that if it's not a `SINK` it's a `SOURCE` (see `ConnectorType.UNKNOWN`). 

I'm also pretty sure that swapping classloaders is not required (as in the case where `AbstractHerder`'s `getConnector` is called, but we'd have to double check.